### PR TITLE
Exactly once automated testing

### PR DESCRIPTION
**Pull request by Paultagoras**
## Checklist
Delete items not relevant to your PR:
- [ ] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

**Comment by mshustov** [comment]
Could you add instructions on running the tests?

File: src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java [hunk 1]
```diff
@@ -412,9 +419,9 @@ public void doInsertRawBinary(List<Record> records) throws IOException, Executio
             CompletableFuture<ClickHouseResponse> future;
 
             try (ClickHousePipedOutputStream stream = ClickHouseDataStreamFactory.getInstance()
-                    .createPipedOutputStream(config, null)) {
+                    .createPipedOutputStream(config)) {
                 // start the worker thread which transfer data from the input into ClickHouse
-                future = request.data(stream.getInputStream()).send();
+                future = request.data(stream.getInputStream()).execute();
```
Comment (mzitnik) [inline src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java:424]: What is the difference between send and execute

File: src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java [hunk 2]
```diff
@@ -488,9 +495,9 @@ public void doInsertJson(List<Record> records) throws IOException, ExecutionExce
...
```
Comment (mzitnik) [inline src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java:500]: Same question here

File: src/main/java/com/clickhouse/kafka/connect/sink/state/provider/KeeperStateProvider.java [hunk 3]
```diff
@@ -1,6 +1,8 @@
 package com.clickhouse.kafka.connect.sink.state.provider;
 
 import com.clickhouse.client.*;
+import com.clickhouse.data.ClickHouseFormat;
+import com.clickhouse.data.ClickHouseRecord;
```
Comment (mzitnik) [inline src/main/java/com/clickhouse/kafka/connect/sink/state/provider/KeeperStateProvider.java:5]: where is that used in the code?

File: src/test/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkTaskSchemalessTest.java [hunk 4]
```diff
@@ -21,7 +21,7 @@ public class ClickHouseSinkTaskSchemalessTest {
     private static ClickHouseHelperClient chc = null;
 
     @BeforeAll
-    private static void setup() {
+    public static void setup() {
```
Comment (mzitnik) [inline src/test/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkTaskSchemalessTest.java:24]: public can be also removed according https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall

File: src/test/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkTaskWithSchemaTest.java [hunk 5]
```diff
@@ -32,7 +32,7 @@ public class ClickHouseSinkTaskWithSchemaTest {
     private static ClickHouseHelperClient chc = null;
 
     @BeforeAll
-    private static void setup() {
+    public static void setup() {
```
Comment (mzitnik) [inline src/test/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkTaskWithSchemaTest.java:35]: same here
public can be also removed according https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall

**Pull request merged**